xbmc/utils/DatabaseUtils.h

```diff
@@ -171,6 +171,7 @@ class DatabaseUtils
 
   static std::string BuildLimitClause(int end, int start = 0);
   static std::string BuildLimitClauseOnly(int end, int start = 0);
+  static unsigned int GetLimitCount(int end, int start);
```

Comment:
I would probably also return `size_t` here since you pass the result to `reserve()`.